src/librustdoc/config.rs

```diff
@@ -446,12 +448,22 @@ impl Options {
         }
 
         // check for `--output-format=json`
-        if !matches!(matches.opt_str("output-format").as_deref(), None | Some("html"))
+        if let Some(format) = matches.opt_str("output-format").as_deref()
+            && format != "html"
             && !matches.opt_present("show-coverage")
```

Comment:
What happens if I run `rustdoc --output-format=doctest --show-coverage`

Reply:
Good point! Gonna also add a UI test for this case.